integration-tests/src/fees.rs

```diff
@@ -40,10 +87,103 @@ fn nft_mint_transaction_fees() {
+
+		// Min fee adjustment multiplier
+		pallet_transaction_payment::pallet::NextFeeMultiplier::<basilisk_runtime::Runtime>::put(
+			basilisk_runtime::MinimumMultiplier::get(),
+		);
+
+		let min_multiplier_rust_fees = TransactionPayment::compute_fee(rust_encoded_len, &info, 0);
+		let min_multiplier_ui_fees = TransactionPayment::compute_fee(ui_encoded_len, &info, 0);
+
+		println!(
+			"NFT mint:\n\t UI fees: {}/{} [actual/expected]\n\t Rust encoded fees: {}/{} [actual/expected]\n\t Fees with min. FeeMultiplier: {} [UI], {} [Rust]",
+			format_num(ui_fees * 10_000 / UNITS, 4),
+			format_num(expected_ui_fees * 10_000 / UNITS, 4),
+			format_num(rust_encoded_fees * 10_000 / UNITS, 4),
+			format_num(expected_rust_encoded_fees * 10_000 / UNITS, 4),
+			format_num(min_multiplier_ui_fees * 10_000 / UNITS, 4),
+			format_num(min_multiplier_rust_fees * 10_000 / UNITS, 4),
+		);
 	});
 }
+
+#[test]
+fn transaction_fees_should_be_as_expected_when_nft_class_is_created() {
+	Basilisk::execute_with(|| {
+		//NOTE: Price showed by polkadotAPPS is changing at second decimal place between runs.
+		let diff = UNITS / 10; //0.1
+
+		let expected_rust_encoded_fees = 39_879 * UNITS / 100; //398.79
+		let expected_ui_fees = 39_982 * UNITS / 100; //399.82
+
+		let call = pallet_nft::Call::<basilisk_runtime::Runtime>::create_class {
+			class_id: 0,
+			class_type: ClassType::Marketplace,
+			metadata: b"ipfs://QmQu2jUmtFNPd86tEHFs6hmAArKYyjEC3xuwVWpFGjcMgm"
+				.to_vec()
+				.try_into()
+				.unwrap(),
+		};
+
+		let info = call.get_dispatch_info();
+		//rust encoded fees
+		let rust_encoded_len = call.encoded_size() as u32;
+		let rust_encoded_fees = TransactionPayment::compute_fee(rust_encoded_len, &info, 0);
+
+		assert_eq_approx!(
+			rust_encoded_fees,
+			expected_rust_encoded_fees,
+			diff,
+			"Rust encoded fees are"
+		);
+
+		//UI encoded fees
+		let ui_encoded_len = 177; //UI encoded length
+		let ui_fees = TransactionPayment::compute_fee(ui_encoded_len, &info, 0);
+
+		assert_eq_approx!(ui_fees, expected_ui_fees, diff, "UI encoded fees are");
+
+		// Min fee adjustment multiplier
+		pallet_transaction_payment::pallet::NextFeeMultiplier::<basilisk_runtime::Runtime>::put(
+			basilisk_runtime::MinimumMultiplier::get(),
+		);
+
+		let min_multiplier_rust_fees = TransactionPayment::compute_fee(rust_encoded_len, &info, 0);
+		let min_multiplier_ui_fees = TransactionPayment::compute_fee(ui_encoded_len, &info, 0);
+
+		println!(
+			"NFT create_class\n\t UI fees: {}/{} [actual/expected]\n\t Rust encoded fees: {}/{} [actual/expected]\n\t Fees with min. FeeMultiplier: {} [UI], {} [Rust]",
+			format_num(ui_fees * 10_000 / UNITS, 4),
+			format_num(expected_ui_fees * 10_000 / UNITS, 4),
+			format_num(rust_encoded_fees * 10_000 / UNITS, 4),
+			format_num(expected_rust_encoded_fees * 10_000 / UNITS, 4),
+			format_num(min_multiplier_ui_fees * 10_000 / UNITS, 4),
+			format_num(min_multiplier_rust_fees * 10_000 / UNITS, 4),
+		);
```

Comment:
dumb question: will this ever be printed?
if the asserts fail, this code won't execute, right?
and if no assert fails, then output is not printed

do you expect to execute with `--no-capture` when interested in the print-out?

Reply:
yes, you are right. 
I expect to use `--no-capture` if you are interested to see values with more precision or if you are setting "new fees" you can temporally comment asserts and look into values(you don't have to check it in the UI) and update expected values last.

Reply:
but true, asserts can be done after printing and commenting will not be necessary, I'll do it